packages/platform-sdk-ada/src/services/client.ts

```diff
@@ -85,7 +91,37 @@ export class ClientService implements Contracts.ClientService {
 	}
 
 	public async wallet(id: string): Promise<Contracts.WalletData> {
-		return new WalletData(await this.get(`v2/wallets/${id}`));
+		const usedSpendAddresses: Set<string> = new Set<string>();
+		const usedChangeAddresses: Set<string> = new Set<string>();
+
+		let offset = 0;
+		let exhausted: boolean = false;
+		do {
+			const spendAddresses: string[] = await this.addressesChunk(id, false, offset);
+			const changeAddresses: string[] = await this.addressesChunk(id, true, offset);
+
+			const allAddresses = spendAddresses.concat(changeAddresses);
+			const usedAddresses: string[] = await this.fetchUsedAddressesData(allAddresses);
+
+			spendAddresses
+				.filter((sa) => usedAddresses.find((ua) => ua === sa) !== undefined)
+				.forEach((sa) => usedSpendAddresses.add(sa));
+			changeAddresses
+				.filter((sa) => usedAddresses.find((ua) => ua === sa) !== undefined)
+				.forEach((sa) => usedChangeAddresses.add(sa));
+
+			exhausted = usedAddresses.length === 0;
+			offset += 20;
+		} while (!exhausted);
```

Comment:
I know this iteration looks ridiculous, but I think it's necessary. Doing multiple queries seem to be the only way to find out which addresses are used and which aren't. Once we have that (after multiple iterations) we can fetch the utxos and by aggregating we calculate the account balance.